pkg/skaffold/test/custom/custom_test.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package custom
+
+import (
+	"context"
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestNewCustomTestRunner(t *testing.T) {
+	testutil.Run(t, "Testing new custom test runner", func(t *testutil.T) {
+		tmpDir := t.NewTempDir().Touch("test.yaml")
+
+		custom := latest.CustomTest{
+			Command:        "echo Running Custom Test command.",
+			TimeoutSeconds: 10,
+			Dependencies: &latest.CustomTestDependencies{
+				Command: "echo [\"file1\",\"file2\",\"file3\"]",
+				Paths:   []string{"**"},
+				Ignore:  []string{"b*"},
+			},
+		}
+
+		cfg := &mockConfig{
+			workingDir: tmpDir.Root(),
+			tests: []*latest.TestCase{{
+				ImageName:      "image",
+				StructureTests: []string{"test.yaml"},
+				CustomTests:    []latest.CustomTest{custom},
+			}},
+		}
+
+		testRunner, err := New(cfg, cfg.workingDir, custom)
+		t.CheckNoError(err)
+		err = testRunner.Test(context.Background(), ioutil.Discard, nil)
```

Comment:
Please use `test.Override` and not actually rely on executing the command in the unit test.
We want to limit the the scope of unit test to
1) making sure the `util.RunCmd` is called with correct parameters.
2) The function does the right thing based on your mock return value.

Reply:
This design is similar to both structure tests & the main test.

Reply:
I am not talking about the mock config. I am talking about the command in test cases "echo" etc actually being executed. They should not be executed via mocking the `util.RunCmd` method.

Reply:
Updated tests to use mock by overriding util.RunCmd method.